### ISSUE TITLE
AudioPlayer::Soundfile::soundfile_error

### ISSUE THREAD

**Issue by RarogCmex** [SoundScapeRenderer/ssr#154]
SSR 0.5 

terminate called after throwing an instance of 'AudioPlayer::Soundfile::soundfile_error'
  what():  get_format(): ERROR: Can't connect; chainsetup not valid!
Cannot read socket fd = 5 err = Выполнено
CheckRes error
JackSocketClientChannel read fail
terminate called recursively
Unable to start Dr. Konqi

**Comment by RarogCmex** [comment]
That's with any file.
SSR being compiled and istalled without issues

**Comment by JensAhrens** [comment]
What kind of sound files is it?

**Comment by JensAhrens** [comment]
... I could also imagine that there is an issue with the path to the file (special characters or paths that are too long or so). Could this be?

**Comment by RarogCmex** [comment]
Tested with wav and flac files.
> ... I could also imagine that there is an issue with the path to the file (special characters or paths that are too long or so). Could this be?
Yes, it could, because my native language is Russian so my system uses ru_RU.utf-8 locale. I tested it with paths without any cirrylic symbols but it didn't help so I haven't had any confidence yet.

**Comment by RarogCmex** [comment]
Example of tested stuff
[Test_Epidemia_Opus.zip](https://github.com/SoundScapeRenderer/ssr/files/3562628/Test_Epidemia_Opus.zip)

**Comment by JensAhrens** [comment]
Sorry for the delay... I tested your file on maxOS and it works well. The only proposition that I have is re-installing libsndfile and ecasound. We've seen problems with ecasound on some systems that could only be solved by compiling ecasound on that system. 

Your error sounds similar to this one: https://ssr.readthedocs.io/en/0.5.0/operation.html#ecasound-cannot-open-a-jack-port (there is a solution proposed)

Finally, another issue can be that jack might be running with realtime priority. This can prevent clients without realtime priority from connecting to it.

**Comment by RarogCmex** [comment]
You're never guilty in the unavoidable :)
That's seems strange, because my Gentoo Linux is source-based and I set up that any package with jack support compiled with its (with added USE="jack" in global make.conf). But I'll test Ecasound connections.
The issue with realtime jack... I didn't use to see that before, And I didn't think that it's possible.

**Comment by JensAhrens** [comment]
Please check if ecasound can play your files to jack. The command would be something like 

`ecasound -i AUDIO_FILE_NAME.FLAC -o jack,system` 

Make sure you keep the volume low. ;) It'll play with maximum amplitude directly to your audio interface.

**Comment by RarogCmex** [comment]
Yes, Ecasoud seems working well. I hearing my file without any distortion
`**************************************************************************
*        ecasound v2.9.1 (C) 1997-2014 Kai Vehmanen and others    
**************************************************************************
(audioio_jack_manager) JACK transport: send/receive disabled (mode: notransport)
(eca-chainsetup) Chainsetup "untitled-chainsetup"
(eca-chainsetup) "rt" buffering mode selected.
(eca-chainsetup) Opened input "Test_Europe_Carry.wav", mode "read". Format: s16_le, channels 2, srate 44100, interleaved (locked
... params).
(eca-chainsetup) NOTE: using existing audio parameters -f:f32_le,2,44100 for object 'jack' (tried to open with
... -f:s16_le,2,44100).
(eca-chainsetup) Opened output "jack", mode "write". Format: f32_le, channels 2, srate 44100, noninterleaved (locked params).
- [ Connected chainsetup: "untitled-chainsetup" ] ------------------------
- [ Controller/Starting batch processing ] -------------------------------
- [ Engine - Driver start ] ----------------------------------------------
^C- [ Controller/Batch processing finished (0) ] ---------------------------
- [ Engine exiting ] -----------------------------------------------------
(eca-control-objects) Disconnecting chainsetup:  "untitled-chainsetup".
`

**Comment by JensAhrens** [comment]
Sorry, I have no idea what to do now. The problem occurs [here](https://github.com/SoundScapeRenderer/ssr/blob/177b6e3ec64ea6c081b9b020db3d4577a7a889c8/src/audioplayer.cpp#L130). 

Ecasound seems to be having trouble opening the file. I'd suggest that you try to invoke the same ecasound commands on the command line that are being invoked in lines 120-130 and see if that tells you anything...
